File: HW7/151044058_CSE241_HW7/Circle.java
import java.awt.Color;
import java.awt.Graphics;

import javax.swing.JPanel;
/**
 * 
 * @author Ali Haydar Kurban
 *
 */
public class Circle extends JPanel implements Shape 
{
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	/**
	 * Raius degiskeni
	 */
	private	double radius;
	/**
	 * center_x degiskeni
	 */
	private double center_x;
	/**
	 * center_y degiskeni
	 */
	private double center_y;
	/**
	 * type degiskeni buyuk sekil kucuk sekil ayirt edebilmek icin
	 */
	String type;

	
	/**
	 * No parameter constructor
	 */
	public Circle()
	{
		radius = 500.0; 
		center_x = 500.0;
		center_y = 500.0;
	}
	
	/**
	 * Gelen radius degerini privatetaki radiusa atar
	 * @param temp_radius radius a atamak yapmak icin kullanilir
	 */
	public Circle(double temp_radius)
	{
		//Throw excepttion
		if(temp_radius <=0 )
			throw new IllegalArgumentException("Illegal Number for Circle\n");
		else
			radius = temp_radius;
	}
	
	/**
	 * Gelen degerleri private degiskenlere atar
	 * @param temp_radius radius a atama yapmak icin kullanilir
	 * @param cX center_x atama yapmak icin kullanilir
	 * @param cY center_y atama yapmak icin kullanilir
	 */
	public Circle(double temp_radius, double cX, double cY)
	{
		//Throw excepttion
		if(temp_radius <= 0 || cX < 0 || cY < 0)
		{
			throw new IllegalArgumentException("Illegal Number for Circle\n");
		}
		else
		{
			radius = temp_radius;
			center_x = cX;
			center_y = cY;	
		}
		
	}
	
	/**
	 * 
	 * @return radius  u return eder
	 */
	public double getRadius()
	{
		return radius; 
	}
	/**
	 * 
	 * @return center_x return eder
	 */
	public double getCenter_x()
	{
		return center_x;
	}
	
	/**
	 * 
	 * @return center_y return eder
	 */
	public double getCenter_y()
	{
		return center_y;
	}
	
	/**
	 * 
	 * @param radiusValue radius' u set etme
	 */
	public void setRadius(double radiusValue)
	{
		radius = radiusValue; 
	}
	
	/**
	 * 
	 * @param center_xValue conter_x' i set etme
	 */
	public void setCenter_x(double center_xValue)
	{
		center_x = center_xValue;
	}
	
	/**
	 * 
	 * @param center_yValue center_y' i set etme
	 */
	public void setCenter_y(double center_yValue)
	{
		center_y = center_yValue;
	}

	/**
	 * Cevre fonksiyonu
	 */

	@Override
	public double perimeter()
	{
		return(2 * Math.PI * radius);
	}
	
	/**
	 * Alan fonksiyonu
	 */
	@Override
	public double area()
	{
		return(Math.PI * radius * radius);
	}
	
	/**
	 * Koordinatlari arttirma
	 */
	@Override
	public Shape increment()
	{
		center_x = center_x + 1;
		center_y = center_y + 1;
		return this;
	}
	
	/**
	 * Koordinatlari azaltma
	 */
	@Override
	public Shape decrement()
	{
		center_x = center_x - 1;
		center_y = center_y - 1;
		return this;
	}
	
	/**
	 * 
	 * @return type return eder(buyuk sekil kucuk sekil icin)
	 */
	public String getType()
	{
		return type;
	}
	
	/**
	 * 
	 * @param typeValue tyope' a set etme icin kullanilir
	 */
	public void setType(String typeValue)
	{
		type = typeValue;
	}
	
	/**
	 * Circle cizdirme islemi
	 */
	@Override
	public void draw(Graphics g)
	{
		if(this.getType() == "CIRCLE") //Buyuk sekil cizdirirken renk beklirmek icin
		{
			g.setColor(Color.RED);
	        g.fillOval((int)this.getCenter_x(),(int)this.getCenter_y(),(int)this.getRadius()*2,(int)this.getRadius()*2);
		}
		if(this.getType() == "circle") //Buyuk sekil cizdirirken renk beklirmek icin
		{
			g.setColor(Color.GREEN);
	        g.fillOval((int)this.getCenter_x(),(int)this.getCenter_y(),(int)this.getRadius()*2,(int)this.getRadius()*2);
		}
	}
	
	/**
	 * JPanle in override edilmis fonksiyonu draw fonksiyonunu cagiririz
	 */
	@Override
	public void paintComponent(Graphics g)
	{
		super.paintComponent(g);
		draw(g);
	}

	/**
	 * Alanlarin kiyaslanmasina gore compareTo fonksiyonunun override hali
	 */
	@Override
	public int compareTo(Object arg0) {
		if(((Shape) arg0).area() > this.area())
			return -1;
		if(((Shape) arg0).area() < this.area())
			return 1;
		
		return 0;
	}
}

File: HW5/151044058_CSE241_HW5/Circle.cpp
#include "Circle.h"

namespace HW5_151044058
{
	Circle :: Circle() : radius(500), center_x(500), center_y(500)
	{ 
		static_total_area = static_total_area + area(); //Static alani arttirma
		static_total_PL = static_total_PL + perimeter(); // Static Cevreyi arttirma
	}

	Circle :: Circle(double temp_radius)throw(Circle_Error)
	{
		//Radius un negatif gelmesi durumunda throw yaptim
		if(temp_radius <= 0)
			throw Circle_Error();
		radius = temp_radius;
		static_total_area = static_total_area + area(); //Static alani arttirma
		static_total_PL = static_total_PL + perimeter();	// Static Cevreyi arttirma
	}

	Circle :: Circle(double temp_radius, double cX, double cY)throw(Circle_Error)
	{
		//Radiusun ve koordinatlarin hatali gelemsi durumunda throw yaptim
		if(temp_radius <= 0 || cX < 0 || cY < 0 ||
			cX > temp_radius || cY > temp_radius)
			throw Circle_Error();
		radius = temp_radius;
		center_x = cX;
		center_y = cY;
		static_total_area = static_total_area + area(); //Static alani arttirma
		static_total_PL = static_total_PL + perimeter();// Static Cevreyi arttirma
	}

	inline double Circle :: getRadius()const { return radius; }
	inline double Circle :: getCenter_x()const { return center_x; }
	inline double Circle :: getCenter_y()const { return center_y; }

	void Circle :: setRadius(double radiusValue) 
	{ 
		radius = radiusValue; 
	}
	void Circle :: setCenter_x(double center_xValue) 
	{ 
		center_x = center_xValue; 
	}
	void Circle :: setCenter_y(double center_yValue) 
	{ 
		center_y = center_yValue; 
	}
	string Circle :: getType()const
	{
		return type;
	}
	void Circle :: setType(string typeValue)
	{
		type = typeValue;
	}
	double Circle :: perimeter()
	{
		const double PI = M_PI;
		return(2 * PI * radius);
	}
	double Circle :: area()
	{
		const double PI = M_PI;
		return(PI * radius * radius);
	}
	void Circle :: draw_Shape(ostream& file)
	{
		if(this->getType() == "CIRCLE")//Buyuk sekillleri kirmizi yazdirmak icin
		{
			file << "<circle cx=" << "\"" << this->getCenter_x() << "\"" << " cy=" << "\"" <<  this->getCenter_y() << "\"" << " r="
				 << "\"" << this->getRadius() << "\"" <<  " fill=\"red\" stroke=\"white\" />\n";
		}
		else if(this->getType() == "circle")//Kucuk sekillleri yesil yazdirmak icin
		{
			file << "<circle cx=" << "\"" << this->getCenter_x() << "\"" << " cy=" << "\"" <<  this->getCenter_y() << "\"" << " r="
				 << "\"" << this->getRadius() << "\"" <<  " fill=\"green\" stroke=\"white\" />\n";
		}

	}

	Shape& Circle :: operator++() //Prefix
	{
		center_x = center_x + 1;
		center_y = center_y + 1;
		return *this;
	}
	Shape& Circle :: operator++(int ignore) //Postfix
	{
		double center_x_temp = center_x;
		double center_y_temp = center_y;
		center_x_temp = center_x_temp + 1;
		center_y_temp = center_y_temp + 1;
		center_x = center_x_temp;
		center_y = center_y_temp;

		return *this;
	}

	Shape& Circle :: operator--() //Prefix
	{
		center_x = center_x - 1;
		center_y = center_y - 1;
		return *this;
	}
	Shape& Circle :: operator--(int ignore) //Postfix
	{
		double center_x_temp = center_x;
		double center_y_temp = center_y;
		center_x_temp = center_x_temp - 1;
		center_y_temp = center_y_temp - 1;
		center_x = center_x_temp;
		center_y = center_y_temp;

		return *this;
	}

	const Circle Circle :: operator+(double bigger)const // Radiusa double ekler ve objeyi return eder
	{
		return(Circle((radius + bigger),center_x,center_y));
	}

	const Circle Circle :: operator-(double smaller)const	// Radiusdan double cikarir ve objeyi return eder
	{
		return(Circle((radius - smaller),center_x,center_y));
	}

	double Circle :: return_total_area() //Static toplam alani return eder
	{
		return static_total_area;
	}
	double Circle :: return_total_PL() //Static toplam cevreyi return eder
	{
		return static_total_PL;
	}
}